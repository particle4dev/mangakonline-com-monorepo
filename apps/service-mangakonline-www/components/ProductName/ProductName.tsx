import * as React from 'react';
import ClassNames from 'classnames';
import { WithStyles, createStyles, withStyles } from '@material-ui/core';
import { ROUTER } from '../../constants';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const debug = require('debug')('components:ProductName');

const styles = () => createStyles({
  root: {
    display: 'flex',
    flex: '0 0 auto',
    height: '24px',
    margin: '20px 0'
  },

  productName: {
    display: 'inline-block',
    font: '24px/24px Roboto, sans-serif',
    letterSpacing: 0,
    verticalAlign: 'top',
    whiteSpace: 'nowrap'
  },

  gutterLeft: {
    margin: '0 16px'
  }
});

type ProductNameProps = WithStyles<typeof styles> & {
  gutterLeft?: boolean,
  title?: string
}

const ProductName = React.forwardRef(function ProductName(props: ProductNameProps, ref: React.Ref<HTMLDivElement>) {
  debug('render');

  const { gutterLeft, classes, title, ...other } = props;
  const clsProductName = ClassNames(classes.productName, {
    [classes.gutterLeft]: gutterLeft
  });

  return (
    <div ref={ref} className={classes.root} {...other}>
      <a
        href={ROUTER.HOME.path}
        data-label="Site Name"
      >
        <span className={clsProductName}>{title}</span>
      </a>
    </div>
  );
});

if (process.env.NODE_ENV !== 'production') {
  ProductName.displayName = 'components__ProductName';
}

ProductName.defaultProps = {
  gutterLeft: false,
  title: process.env.NEXT_PUBLIC_SITE_NAME
};

export default React.memo(withStyles(styles, {name: 'ProductName'})(ProductName));
