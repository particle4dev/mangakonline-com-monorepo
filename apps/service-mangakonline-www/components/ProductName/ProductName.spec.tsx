import * as React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ProductName from './ProductName';

describe('ProductName', () => {
  it('should render successfully', () => {
    const { baseElement, getByText } = render(<ProductName title="test" />);
    expect(getByText('test')).toBeInTheDocument();
    // Snapshot demo
    expect(baseElement).toMatchSnapshot();
  });
});
