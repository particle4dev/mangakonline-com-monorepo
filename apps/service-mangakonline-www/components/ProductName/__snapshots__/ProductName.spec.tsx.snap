// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductName should render successfully 1`] = `
<body>
  <div>
    <div
      class="ProductName-root-1"
    >
      <a
        data-label="Site Name"
        href="/"
      >
        <span
          class="ProductName-productName-2"
        >
          test
        </span>
      </a>
    </div>
  </div>
</body>
`;
